.github/workflows/run-pytest.yml

```
@@ -0,0 +1,66 @@
+name: Run Unit Tests
+
+on:
+  # trigger on pull requests
+  pull_request:
+
+  # trigger on all commits to master
+  push:
+    branches:
+      - 'master'
+
+  # trigger on request
+  workflow_dispatch:
+
```

```suggestion
concurrency:
  group: "${{ github.workflow }}-${{ github.ref }}"
  cancel-in-progress: true
```

Something like this ^  will kill the workflow when another commit is made, instead of running the workflow twice. https://docs.github.com/en/actions/using-jobs/using-concurrency#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow
I like it since it helps reduce CI waste when someone makes a quick commit after pushing to fix some minor thing.